### ISSUE TITLE
Volume headers errors

### ISSUE THREAD

**Issue by jadekler** [concourse/concourse#1405]
# Bug Report

* Concourse version: 3.3.1
* Deployment type (BOSH/Docker/binary): binary
* Infrastructure/IaaS: AWS
* Browser (if applicable): chrome
* Did this used to work? Yes

We get the following volume headers error when a task attempts to execute, intermittently:

```
Post /volumes: net/http: timeout awaiting response headers
```

Our setup:

- 2 workers with 16gb RAM and 4 cores, 1 web with 8gb and 4 cores (IIRC)
- Postgres db lives on the web VM
- Workers and web never get above 10% CPU
- Three pipelines with < 5 jobs, 1 pipeline with ~20 jobs, some of which have 3000+ builds (most are low hundreds though)

![screen shot 2017-07-20 at 11 12 55](https://user-images.githubusercontent.com/3584893/28430082-6b2fb1b2-6d3c-11e7-9e81-4cac5fcc19e5.png)

**Comment by krishicks** [comment]
I didn't see this particular error.

**Comment by jadekler** [comment]
@krishicks Thanks for the note, I'll remove you from the original post. I can't remember who else saw it... maybe I'll find them again later.

**Comment by smgoller** [comment]
We've been having similar problems, and we "resolved" the issue by launching an instance with a gp2 root drive instead of standard. That implies to me that there's a performance issue with fetching and deploying docker images with baggageclaim and it's hitting the volume timeout.

**Comment by krishicks** [comment]
> Infrastructure/IaaS: AWS

Is this maybe due to BurstBalance on AWS?

See https://jeremyeder.com/2017/07/25/docker-operations-slowing-down-on-aws-this-time-its-not-dns/.

**Comment by dgodd** [comment]
I've been seeing this on GCP. An example https://buildpacks.ci.cf-app.com/teams/main/pipelines/binary-builder/jobs/build-php/builds/39

**Comment by drahnr** [comment]
I have the same issue with `3.4.0`, one worker with one worker (16g RAM / 8 cores / `docker`ized everything, `ext4`/`overlayfs`)
![screenshot from 2017-08-31 06-47-59](https://user-images.githubusercontent.com/667047/29908155-39abaa74-8e20-11e7-9783-3a2cedeea293.png)

Every single of my jobs fails with
```
Post /volumes: net/http: timeout awaiting response headers
```
At the push step.

**Comment by drahnr** [comment]
@vito this should be closed since 3.4.1

**Comment by vito** [comment]
Yep! Thanks.
